mem/ctrl/ContextSwitchRTL.py

```diff
@@ -0,0 +1,91 @@
+"""
+==========================================================================
+ContextSwitchRTL.py
+==========================================================================
+Records/resumes progress (itertion) for functional unit (FU).
+
+Author : Yufei Yang
+  Date : Aug 11, 2025
+"""
+from pymtl3.stdlib.primitive import *
+from ...lib.basic.val_rdy.ifcs import ValRdyRecvIfcRTL as RecvIfcRTL
+from ...lib.basic.val_rdy.ifcs import ValRdySendIfcRTL as SendIfcRTL
+from ...lib.cmd_type import *
+from ...lib.opt_type import *
+from ...lib.status_type import *
+from ...lib.util.common import *
+
+class ContextSwitchRTL(Component):
+
+  def construct(s, data_nbits):
+
+    # Constant
+    CmdType = mk_bits(clog2(NUM_CMDS))
+    StatusType = mk_bits(clog2(NUM_STATUS))
+    DataType = mk_bits(data_nbits)
+    OptType = mk_bits(clog2(NUM_OPTS))
+
+    # Interface
+    s.recv_cmd = InPort(CmdType)
+    s.recv_cmd_vld = InPort(b1)
+    s.recv_opt = InPort(OptType)
+    s.progress_in = InPort(DataType)
+    s.progress_in_vld = InPort(b1)
```

Comment:
Add a comment above mentioning "We don't need recv_opt_vld as progress_in_vld is enough to tell whether the opt is valid"?

And this PR is ready for review?